README.md
# Form Page View

Form Page View is a Flutter package that allows you to create a form with a page view and a progress bar. This package makes it easy to create interactive and customizable forms with seamless navigation between pages.

## Demo

| Circular indicator | Linear indicator |
|-------|---|
| ![Example screenshot](assets/Screenshot_2023-05-20-22-05-03-030_com.example.form_page_view_test.jpg) | ![Example screenshot](assets/Screenshot_2023-05-20-22-04-51-028_com.example.form_page_view_test.jpg) |
| ![Example screenshot](assets/Screenrecorder-2023-05-20-22-02-55-536.gif) | ![Example screenshot](assets/Screenrecorder-2023-05-20-22-03-32-538.gif) |


## Platforms supported

| Android | iOS | Web | MacOS | Windows | Linux |
|:-------:|:---:|:---:|:-----:|:-------:|:-----:|
|✅|✅|✅|✅|✅|✅|

## Features

- Create a form with a page view and a progress bar
- Navigate between pages using the progress bar or the back button
- Customize the appearance of the form
- Customize the progress indicator
- Customize the button widget

## Installation

Add the following dependency to your `pubspec.yaml` file:

```yaml
dependencies:
  form_page_view: ^1.0.0
```

Import the package in your Dart file:

```dart
import 'package:form_page_view/form_page_view.dart';
```

## Usage

Create a list of page models (**'FormPageModel'**) to define the pages in your form. Each page model should contain a **'formKey'** for form validation and a **'form'** that represents the page content.

```dart
final List<FormPageModel> pages = [
  FormPageModel(
    formKey: GlobalKey<FormState>(),
    body: YourFormWidget(),
    title: 'Page 1',
    textButton: 'Next',
  ),
  // Add other page models here
];
```

Next, create a **PageController** to control the navigation between pages:

```dart
final PageController _pageController = PageController();
```

Use the **'FormPageView'** widget by providing it with the pages, the function to execute when the form is submitted, and other optional parameters to customize the appearance and behavior of the form:

```dart
FormPageView(
  progress: ProgressIndicatorType.linear,
  pages: pages,
  onFormSubmitted: () {
    // Function to execute when the form is submitted
  },
  controller: _pageController,
),
```

Refer to the provided usage **'examples'** in the example folder for more details.

## Properties

| Property | Type | Description |
|-----------|------|-------------|
| progress | ProgressIndicatorType? | Optional. Specifies the type of progress indicator to use. Default is `ProgressIndicatorType.linear`. |
| pages | List\<PageModel> | Required. The list of page models that define the pages in the form. |
| onFormSubmitted | Function() | Required. The function to execute when the form is submitted. |
| controller | PageController | Required. The page controller for controlling the navigation in the pages. |
| contentPadding | EdgeInsetsGeometry? | Optional. The padding of the content of the form pages. Default is `EdgeInsets.all(16)`. |
| buttonPadding | EdgeInsetsGeometry? | Optional. The padding of the page navigation button. Default is `EdgeInsets.all(16)`. |
| progressPadding | EdgeInsetsGeometry? | Optional. The padding of the progress indicator. Default is `EdgeInsets.all(8.0)`. |
| style | FormPageStyle? | Optional. A FormPageStyle object to customize the appearance and behavior of the form. |

## Customization

### Form widget

You can customize the form widget by providing the following parameters:

- **'contentPadding'**: The padding of the content of the form pages. Default is `EdgeInsets.all(16)`.
- **'buttonPadding'**: The padding of the page navigation button. Default is `EdgeInsets.all(16)`.
- **'progressPadding'**: The padding of the progress indicator. Default is `EdgeInsets.all(8.0)`.
- **'style'**: A **FormPageStyle** object to customize the appearance and behavior of the form.

### Form Page Style

You can customize the form page style by providing a **'FormPageStyle'** object to the **'style'** parameter. The following parameters are available:

- **'backgroundColor'**: The background color of the form pages. Default is `Colors.white`.

#### App bar

You can customize the app bar by providing the following parameters:

- **'appBarCenterTitle'**: Specifies whether to center the title on the app bar. Default is `false`.
- **'appBarElevation'**: The elevation of the app bar. Default is `4.0`.
- **'appBarBackgroundColor'**: The background color of the app bar. Default is `Colors.white`.
- **'appBarHeight'**: The height of the app bar. Default is `56.0`.
- **'appBarTextStyle'**: The text style of the app bar title. Default is `TextStyle(color: Colors.black)`.

#### Progress indicator

You can customize the progress indicator by providing the following parameters:

- **'progressIndicatorColor'**: The color of the progress indicator. Default is `Colors.blue`.
- **'progressIndicatorBackgroundColor'**: The background color of the progress indicator. Default is `Colors.grey`.
- **'progressIndicatorSize'**: The size of the progress indicator. Default is `48.0`.
- **'progressIndicatorStrokeWidth'**: The stroke width of the progress indicator. Default is `4.0`.

#### Button widget

You can customize the button widget by providing the following parameters:

- **'buttonStyle'**: The style of the page navigation button. Default is `ElevatedButton.styleFrom(primary: Colors.blue)`.
- **'buttonHeight'**: The height of the page navigation button. Default is `48`.
- **'buttonWidth'**: The width of the page navigation button. Default is `double.infinity`.
- **'buttonTextStyle'**: The text style of the page navigation button. Default is `TextStyle(color: Colors.white)`.
- **'backButtonIcon'**: The icon of the back button. Default is `Icons.arrow_back`.

## Authors

- [@juju149](https://www.github.com/juju149)

CHANGELOG.md
# 0.2.0

* Add `FormPageStyle` class for styling form pages.

# 0.1.1

* Rename class FormPageModel.

# 0.1.0

* Initial development release.



example/main.dart
// ignore_for_file: avoid_print

import 'package:flutter/material.dart';
import 'package:form_page_view/enum/progress_enum.dart';
import 'package:form_page_view/models/form_page_model.dart';
import 'package:form_page_view/form_page_view.dart';
import 'package:form_page_view/models/form_page_style.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        scaffoldBackgroundColor: Colors.white,
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    // Forms keys
    final GlobalKey<FormState> formKeyPage1 = GlobalKey<FormState>();
    final GlobalKey<FormState> formKeyPage2 = GlobalKey<FormState>();
    final GlobalKey<FormState> formKeyPage3 = GlobalKey<FormState>();

    // Controllers
    final emailController = TextEditingController();
    final usernameController = TextEditingController();
    final firstNameController = TextEditingController();
    final lastNameController = TextEditingController();

    // Create a list of FormPageModel objects representing the form pages
    final pages = [
      FormPageModel(
        formKey: formKeyPage1,
        title: 'Page 1',
        textButton: 'Next',
        body: Form(
          key: formKeyPage1,
          child: Column(
            children: [
              const SizedBox(height: 20),
              TextFormField(
                controller: usernameController,
                decoration: const InputDecoration(labelText: 'Username'),
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Please enter a username';
                  }
                  return null;
                },
              )
            ],
          ),
        ),
      ),
      FormPageModel(
        formKey: formKeyPage2,
        title: 'Page 2',
        textButton: 'Next',
        body: Form(
          key: formKeyPage2,
          child: Column(
            children: [
              const SizedBox(height: 20),
              TextFormField(
                controller: firstNameController,
                decoration: const InputDecoration(labelText: 'First Name'),
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Please enter your first name';
                  }
                  return null;
                },
              ),
              const SizedBox(height: 20),
              TextFormField(
                controller: lastNameController,
                decoration: const InputDecoration(labelText: 'Last Name'),
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Please enter your last name';
                  }
                  return null;
                },
              ),
            ],
          ),
        ),
      ),
      FormPageModel(
        formKey: formKeyPage3,
        title: 'Page 3',
        textButton: 'Next',
        body: Form(
          key: formKeyPage3,
          child: Column(
            children: [
              const SizedBox(height: 20),
              TextFormField(
                controller: emailController,
                decoration: const InputDecoration(labelText: 'Email'),
                validator: (value) {
                  if (value == null || value.isEmpty) {
                    return 'Please enter your email';
                  }
                  return null;
                },
              ),
            ],
          ),
        ),
      ),
    ];

    const style = FormPageStyle(
      backgroundColor: Colors.white,
      buttonHeight: 60,
    );

    return FormPageView(
      controller: PageController(),
      progress: ProgressIndicatorType.circular,
      pages: pages,
      style: style,
      onFormSubmitted: () {
        // print all the values
        print('Username: ${usernameController.text}');
        print('First Name: ${firstNameController.text}');
        print('Last Name: ${lastNameController.text}');
        print('Email: ${emailController.text}');
      },
    );
  }
}


lib/pageview_form.dart
// Importation des packages et des fichiers nécessaires pour le widget
library pageview_form;

import 'package:flutter/material.dart';
import 'package:pageview_form/enum/progress_enum.dart';
import 'package:pageview_form/models/page_form_model.dart';

// Définition du widget PageviewForm
class PageviewForm extends StatefulWidget {
  // Paramètres optionnels pour personnaliser l'apparence et le comportement du formulaire
  final ProgressIndicatorType? progress;
  final List<PageFormModel> pages;
  final Function() onFormSubmitted;
  final EdgeInsetsGeometry? padding;
  final Color? backgroundColor;
  final ButtonStyle? buttonStyle;
  final double? buttonHeight;
  final double? buttonWidth;
  final TextStyle? buttonTextStyle;
  final bool? showAppBar;
  final bool? showBackButton;
  final bool? appBarCenterTitle;
  final double? appBarElevation;
  final Color? appBarBackgroundColor;
  final double? appBarHeight;
  final TextStyle? appBarTextStyle;
  final Color? progressIndicatorColor;
  final Color? progressIndicatorBackgroundColor;
  final double? progressIndicatorSize;
  final double? progressIndicatorStrokeWidth;
  final PageController controller;

  const PageviewForm({
    Key? key,
    this.progress = ProgressIndicatorType.linear,
    required this.pages,
    required this.onFormSubmitted,
    this.padding,
    this.buttonStyle,
    this.showAppBar = true,
    this.showBackButton = true,
    this.buttonHeight = 50,
    this.buttonWidth = double.infinity,
    this.buttonTextStyle,
    this.appBarCenterTitle = false,
    this.appBarElevation = 4,
    this.appBarBackgroundColor = Colors.white,
    this.appBarHeight = 56,
    this.progressIndicatorColor,
    this.progressIndicatorBackgroundColor,
    this.appBarTextStyle,
    this.progressIndicatorSize,
    this.progressIndicatorStrokeWidth,
    this.backgroundColor,
    required this.controller,
  }) : super(key: key);

  @override
  _PageviewFormState createState() => _PageviewFormState();
}

// Définition de l'état du widget PageviewForm
class _PageviewFormState extends State<PageviewForm> {
  // Index de la page actuelle
  int _currentPageIndex = 0;

  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    // Suppression du contrôleur de page
    widget.controller.dispose();
    super.dispose();
  }

  // Validation de la page actuelle
  bool _validateCurrentPage() {
    final page = widget.pages[_currentPageIndex];
    if (page.formKey != null) {
      final form = page.formKey!.currentState!;
      if (form.validate()) {
        form.save();
      } else {
        return false;
      }
    }
    return true;
  }

  // Passage à la page suivante
  void _onNextPressed() {
    if (_validateCurrentPage()) {
      setState(() {
        if (_currentPageIndex < widget.pages.length - 1) {
          _currentPageIndex++;
          widget.controller.nextPage(
              duration: const Duration(milliseconds: 300), curve: Curves.ease);
        } else {
          widget.onFormSubmitted();
        }
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    // Construction de l'interface utilisateur du widget
    return Scaffold(
      backgroundColor: widget.backgroundColor ?? Colors.white,
      appBar: widget.showAppBar!
          ? AppBar(
              centerTitle: widget.appBarCenterTitle ?? false,
              elevation: widget.appBarElevation ?? 4,
              backgroundColor: widget.appBarBackgroundColor ?? Colors.white,
              toolbarHeight: widget.appBarHeight ?? 56,
              leading: widget.showBackButton!
                  ? IconButton(
                      onPressed: () {
                        if (_currentPageIndex > 0) {
                          setState(() {
                            _currentPageIndex--;
                            widget.controller.previousPage(
                              duration: const Duration(milliseconds: 300),
                              curve: Curves.ease,
                            );
                          });
                        }
                      },
                      icon: const Icon(
                        Icons.arrow_back,
                        color: Colors.black,
                      ),
                    )
                  : null,
              title: widget.pages[_currentPageIndex].title != null
                  ? Text(
                      widget.pages[_currentPageIndex].title!,
                      style: widget.appBarTextStyle ??
                          const TextStyle(
                            color: Colors.black,
                            fontSize: 18,
                            fontWeight: FontWeight.bold,
                          ),
                    )
                  : null,
              actions: [
                // Indicateur de progression circulaire ou linéaire
                widget.progress == ProgressIndicatorType.circular
                    ? Padding(
                        padding: const EdgeInsets.all(8.0),
                        child: TweenAnimationBuilder<double>(
                          tween: Tween<double>(
                            begin: 0,
                            end: _currentPageIndex / (widget.pages.length - 1),
                          ),
                          duration: const Duration(milliseconds: 300),
                          builder: (context, value, child) {
                            return SizedBox(
                              width: widget.progressIndicatorSize ?? 40,
                              height: widget.progressIndicatorSize ?? 40,
                              child: CircularProgressIndicator(
                                value: value,
                                strokeWidth:
                                    widget.progressIndicatorStrokeWidth ?? 4,
                                color: widget.progressIndicatorColor ??
                                    Theme.of(context).primaryColor,
                                backgroundColor:
                                    widget.progressIndicatorBackgroundColor,
                              ),
                            );
                          },
                        ),
                      )
                    : const SizedBox.shrink(),
              ],
            )
          : null,
      body: Padding(
        padding: widget.padding ?? const EdgeInsets.all(16.0),
        child: Column(
          children: [
            // Indicateur de progression linéaire ou rien
            widget.progress == ProgressIndicatorType.linear
                ? TweenAnimationBuilder<double>(
                    tween: Tween<double>(
                      begin: 0,
                      end: _currentPageIndex / (widget.pages.length - 1),
                    ),
                    duration: const Duration(milliseconds: 300),
                    builder: (context, value, child) {
                      return LinearProgressIndicator(
                        value: value,
                        minHeight: widget.progressIndicatorStrokeWidth ?? 4,
                        color: widget.progressIndicatorColor ??
                            Theme.of(context).primaryColor,
                        backgroundColor:
                            widget.progressIndicatorBackgroundColor ??
                                Colors.grey[300],
                        valueColor: AlwaysStoppedAnimation<Color>(
                          widget.progressIndicatorColor ??
                              Theme.of(context).primaryColor,
                        ),
                      );
                    },
                  )
                : const SizedBox(
                    height: 0,
                  ),
            // PageView pour afficher les pages du formulaire
            Expanded(
              child: PageView.builder(
                controller: widget.controller,
                itemCount: widget.pages.length,
                physics: const NeverScrollableScrollPhysics(),
                itemBuilder: (context, index) {
                  return widget.pages[index].form;
                },
              ),
            ),
            // Bouton pour passer à la page suivante
            SizedBox(
              height: widget.buttonHeight ?? 50,
              width: widget.buttonWidth ?? double.infinity,
              child: ElevatedButton(
                onPressed: _onNextPressed,
                child: Text(
                  widget.pages[_currentPageIndex].textButton ?? 'Next',
                  style: widget.buttonTextStyle ??
                      TextStyle(
                        color: Theme.of(context).primaryColorLight,
                      ),
                ),
                style: widget.buttonStyle ??
                    ElevatedButton.styleFrom(
                      backgroundColor: Theme.of(context).primaryColor,
                    ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}


lib/models/form_page_model.dart
import 'package:flutter/material.dart';

/// Represents a page model for the form.
class FormPageModel {
  /// The title of the page.
  final String? title;

  /// The form widget for this page.
  final Widget body;

  /// The text displayed on the button for this page.
  final String? textButton;

  /// The value indicating whether the button is enabled.
  final bool? isButtonEnabled;

  /// The callback function executed when the button is pressed.
  final VoidCallback? onPressed;

  /// The style applied to the button.
  final ButtonStyle? buttonStyle;

  /// The key for the form.
  final GlobalKey<FormState>? formKey;

  /// Creates a new instance of [FormPageModel].
  ///
  /// [title] (optional) - The title of the page.
  /// [body] (required) - The form widget for this page.
  /// [textButton] (optional) - The text displayed on the button for this page.
  /// [onPressed] (optional) - The callback function executed when the button is pressed.
  /// [buttonStyle] (optional) - The style applied to the button.
  /// [formKey] (optional) - The key for the form.
  FormPageModel({
    this.title,
    required this.body,
    this.textButton,
    this.onPressed,
    this.buttonStyle,
    this.formKey,
    this.isButtonEnabled,
  });
}


lib/models/form_page_style.dart
// Create a class FormPageStyle that containe all properties to custom form page view
import 'package:flutter/material.dart';

class FormPageStyle {
  // Background color of the form page view
  final Color? backgroundColor;

  // Style of the submit button
  final ButtonStyle? buttonStyle;

  // Height of the submit button
  final double? buttonHeight;

  // Width of the submit button
  final double? buttonWidth;

  // Text style of the submit button
  final TextStyle? buttonTextStyle;

  // Whether the app bar title should be centered
  final bool? appBarCenterTitle;

  // Elevation of the app bar
  final double? appBarElevation;

  // Background color of the app bar
  final Color? appBarBackgroundColor;

  // Height of the app bar
  final double? appBarHeight;

  // Text style of the app bar title
  final TextStyle? appBarTextStyle;

  // Color of the progress indicator
  final Color? progressIndicatorColor;

  // Background color of the progress indicator
  final Color? progressIndicatorBackgroundColor;

  // Size of the progress indicator
  final double? progressIndicatorSize;

  // Stroke width of the progress indicator
  final double? progressIndicatorStrokeWidth;

  // Icon of the back button
  final Icon? backButtonIcon;

  // Constructor for the FormPageStyle class
  const FormPageStyle({
    this.backgroundColor,
    this.buttonStyle,
    this.buttonHeight,
    this.buttonWidth,
    this.buttonTextStyle,
    this.appBarCenterTitle,
    this.appBarElevation,
    this.appBarBackgroundColor,
    this.appBarHeight,
    this.progressIndicatorColor,
    this.progressIndicatorBackgroundColor,
    this.appBarTextStyle,
    this.progressIndicatorSize,
    this.progressIndicatorStrokeWidth,
    this.backButtonIcon,
  });
}


lib/form_page_view.dart
// Importation des packages et des fichiers nécessaires pour le widget
library form_page_view;

import 'package:flutter/material.dart';
import 'package:form_page_view/enum/progress_enum.dart';
import 'package:form_page_view/models/form_page_model.dart';
import 'package:form_page_view/models/form_page_style.dart';

// Définition du widget FormPageView
class FormPageView extends StatefulWidget {
  // Paramètres optionnels pour personnaliser l'apparence et le comportement du formulaire
  final ProgressIndicatorType? progress;
  final List<FormPageModel> pages;
  final Function() onFormSubmitted;
  final bool? showAppBar;
  final bool? showBackButton;
  final EdgeInsetsGeometry? contentPadding;
  final EdgeInsetsGeometry? buttonPadding;
  final EdgeInsetsGeometry? progressPadding;
  final PageController controller;
  final FormPageStyle? style;

  const FormPageView({
    Key? key,
    this.progress = ProgressIndicatorType.linear,
    required this.pages,
    required this.onFormSubmitted,
    required this.controller,
    this.contentPadding,
    this.buttonPadding,
    this.progressPadding,
    this.style,
    this.showAppBar,
    this.showBackButton,
  }) : super(key: key);

  @override
  _FormPageViewState createState() => _FormPageViewState();
}

// Définition de l'état du widget FormPageView
class _FormPageViewState extends State<FormPageView> {
  // Index de la page actuelle
  int _currentPageIndex = 0;

  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    // Suppression du contrôleur de page
    widget.controller.dispose();
    super.dispose();
  }

  // Validation de la page actuelle
  bool _validateCurrentPage() {
    final page = widget.pages[_currentPageIndex];
    if (page.formKey != null) {
      final form = page.formKey!.currentState!;
      if (form.validate()) {
        form.save();
      } else {
        return false;
      }
    }
    return true;
  }

  // Passage à la page suivante
  void _onNextPressed() {
    if (_validateCurrentPage()) {
      setState(() {
        if (_currentPageIndex < widget.pages.length - 1) {
          _currentPageIndex++;
          widget.controller.nextPage(
              duration: const Duration(milliseconds: 300), curve: Curves.ease);
        } else {
          widget.onFormSubmitted();
        }
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    // Construction de l'interface utilisateur du widget
    return Scaffold(
      backgroundColor: widget.style?.backgroundColor ?? Colors.white,
      appBar: widget.showAppBar != null
          ? AppBar(
              centerTitle: widget.style?.appBarCenterTitle ?? false,
              elevation: widget.style?.appBarElevation ?? 4,
              backgroundColor:
                  widget.style?.appBarBackgroundColor ?? Colors.white,
              toolbarHeight: widget.style?.appBarHeight ?? 56,
              leading: widget.showBackButton != null
                  ? IconButton(
                      onPressed: () {
                        if (_currentPageIndex > 0) {
                          setState(() {
                            _currentPageIndex--;
                            widget.controller.previousPage(
                              duration: const Duration(milliseconds: 300),
                              curve: Curves.ease,
                            );
                          });
                        }
                      },
                      icon: widget.style?.backButtonIcon ??
                          const Icon(Icons.arrow_back_ios),
                    )
                  : null,
              title: widget.pages[_currentPageIndex].title != null
                  ? Text(
                      widget.pages[_currentPageIndex].title!,
                      style: widget.style?.appBarTextStyle ??
                          const TextStyle(
                            color: Colors.black,
                            fontSize: 18,
                            fontWeight: FontWeight.bold,
                          ),
                    )
                  : null,
              actions: [
                // Indicateur de progression circulaire ou linéaire
                widget.progress == ProgressIndicatorType.circular
                    ? Padding(
                        padding: const EdgeInsets.all(8.0),
                        child: TweenAnimationBuilder<double>(
                          tween: Tween<double>(
                            begin: 0,
                            end: _currentPageIndex / (widget.pages.length - 1),
                          ),
                          duration: const Duration(milliseconds: 300),
                          builder: (context, value, child) {
                            return SizedBox(
                              width: widget.style?.progressIndicatorSize ?? 40,
                              height: widget.style?.progressIndicatorSize ?? 40,
                              child: CircularProgressIndicator(
                                value: value,
                                strokeWidth: widget
                                        .style?.progressIndicatorStrokeWidth ??
                                    4,
                                color: widget.style?.progressIndicatorColor ??
                                    Theme.of(context).primaryColor,
                                backgroundColor: widget
                                    .style?.progressIndicatorBackgroundColor,
                              ),
                            );
                          },
                        ),
                      )
                    : const SizedBox.shrink(),
              ],
            )
          : null,
      body: Column(
        children: [
          // Indicateur de progression linéaire ou rien
          widget.progress == ProgressIndicatorType.linear
              ? Padding(
                  padding: widget.progressPadding ?? const EdgeInsets.all(8.0),
                  child: TweenAnimationBuilder<double>(
                    tween: Tween<double>(
                      begin: 0,
                      end: _currentPageIndex / (widget.pages.length - 1),
                    ),
                    duration: const Duration(milliseconds: 300),
                    builder: (context, value, child) {
                      return LinearProgressIndicator(
                        value: value,
                        minHeight:
                            widget.style?.progressIndicatorStrokeWidth ?? 4,
                        color: widget.style?.progressIndicatorColor ??
                            Theme.of(context).primaryColor,
                        backgroundColor:
                            widget.style?.progressIndicatorBackgroundColor ??
                                Colors.grey[300],
                        valueColor: AlwaysStoppedAnimation<Color>(
                          widget.style?.progressIndicatorColor ??
                              Theme.of(context).primaryColor,
                        ),
                      );
                    },
                  ),
                )
              : const SizedBox(
                  height: 0,
                ),
          // PageView pour afficher les pages du formulaire
          Expanded(
            child: PageView.builder(
              controller: widget.controller,
              itemCount: widget.pages.length,
              physics: const NeverScrollableScrollPhysics(),
              itemBuilder: (context, index) {
                return Padding(
                  padding: widget.contentPadding ?? const EdgeInsets.all(16),
                  child: widget.pages[index].body,
                );
              },
            ),
          ),
          // Bouton pour passer à la page suivante
          Padding(
            padding: widget.buttonPadding ?? const EdgeInsets.all(16),
            child: SizedBox(
              height: widget.style?.buttonHeight ?? 50,
              width: widget.style?.buttonWidth ?? double.infinity,
              child: ElevatedButton(
                onPressed: _onNextPressed,
                child: Text(
                  widget.pages[_currentPageIndex].textButton ?? 'Next',
                  style: widget.style?.buttonTextStyle ??
                      TextStyle(
                        color: Theme.of(context).primaryColorLight,
                      ),
                ),
                style: widget.style?.buttonStyle ??
                    ElevatedButton.styleFrom(
                      backgroundColor: Theme.of(context).primaryColor,
                    ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
